src/mprk.jl

```diff
@@ -8,13 +8,16 @@ end
 #####################################################################
 p_prototype(u, f) = zeros(eltype(u), length(u), length(u))
 p_prototype(u, f::ConservativePDSFunction) = zero(f.p_prototype)
+p_prototype(u, f::PDSFunction) = zero(f.p_prototype)
+d_prototype(u, f) = zeros(eltype(u), (length(u),))
```

Comment:
I added this possibility to define a `d_prototype`. But now I'm wondering if there is any reason to define such a `d_prototype` at all?  It will probably always be a vector of the same type as `u`. 

I vote in favor of removing it again. What do you think @ranocha ?

Reply:
Sounds good to me - it should just be `similar(u)`. If we need something else later, we can always introduce this additional layer of dispatch